Prowl.Editor/EditorWindows/FileNamingWindow.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Hexa.NET.ImGui;
+using Hexa.NET.ImGuizmo;
+using Prowl.Icons;
+using Prowl.Runtime;
+using Prowl.Runtime.Assets;
+using Silk.NET.GLFW;
+using Silk.NET.Input;
+using System.Reflection;
+
+namespace Prowl.Editor.EditorWindows;
+
+public class FileNamingWindow : EditorWindow
+{
+    public static bool WindowDrawnThisFrame = false;
+
+    private string _searchText = "";
+    private string createName = "";
+
+    protected override ImGuiWindowFlags Flags { get; } = ImGuiWindowFlags.NoCollapse | ImGuiWindowFlags.NoDocking | ImGuiWindowFlags.Tooltip | ImGuiWindowFlags.NoDecoration;
+    protected override bool Center { get; } = true;
+    protected override int Width { get; } = 512;
+    protected override int Height { get; } = 64;
+    protected override bool BackgroundFade { get; } = true;
+
+    DirectoryInfo? Directory;
+    string fileType = "";
+
+    public FileNamingWindow(DirectoryInfo? Directory , string fileType) : base()
+    {
+        Title = FontAwesome6.Book + " Name file";
+        Width = (int)ImGui.GetMousePos().X - 512;
+        Height = -(int)ImGui.GetMousePos().Y + 256;
+        this.Directory = Directory;
+        this.fileType = fileType;
+    }
+
+    protected override void Draw()
+    {
+        WindowDrawnThisFrame = true;
+        ImGui.SetWindowSize(new Vector2(480,32));
+        ImGui.InputText("File name", ref createName, 0x100);
+        ImGui.SameLine();
+        if (ImGui.Button("Save file"))
+        {
+            if (fileType == "Script")
```

Comment:
Instead of hardcoding the types like this, could you pass in an Action to the constructor?
That way this becomes a more generic method that can be reused without changing it.

We could also rename this class to CreateNewFile.cs or something and use it for all file creation

Reply:
Sure, I'll work on that.